plugins/datadog-standalone/datadog-standalone.go

```diff
@@ -0,0 +1,218 @@
+type DDSeries struct {
+	Series []DDMetric `json:"series"`
+}
+
+// A successful metrics API result
+type DDResult struct {
+	Status   string   `json:"status,omitempty"`
+	Errors   []string `json:"errors,omitempty"`
+	Warnings []string `json:"warnings,omitempty"`
+}
+
+func main() {
+	log.Info("Starting Standalone DataDog DC/OS metrics plugin")
+	datadogPlugin, err := plugin.New(
+		plugin.PluginName("standalone-datadog"),
+		plugin.ExtraFlags(pluginFlags),
+		plugin.ConnectorFunc(datadogConnector))
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	log.Fatal(datadogPlugin.StartPlugin())
+}
+
+func messagesToSeries(messages []producers.MetricsMessage) *DDSeries {
+
+	series := new(DDSeries)
+
+	for _, message := range messages {
+		dimensions := &message.Dimensions
+		host := &dimensions.Hostname
+		messageTags := []string{}
+		addMessageTag := func(key, value string) {
+			if len(value) > 0 {
+				messageTags = append(messageTags, fmt.Sprintf("%s:%s", key, value))
+			}
+		}
+
+		for name, value := range dimensions.Labels {
+			addMessageTag(name, value)
+		}
+		addMessageTag("mesosId", dimensions.MesosID)
+		addMessageTag("clusterId", dimensions.ClusterID)
+		addMessageTag("containerId", dimensions.ContainerID)
+		addMessageTag("executorId", dimensions.ExecutorID)
+		addMessageTag("frameworkName", dimensions.FrameworkName)
+		addMessageTag("frameworkId", dimensions.FrameworkID)
+		addMessageTag("frameworkRole", dimensions.FrameworkRole)
+		addMessageTag("frameworkPrincipal", dimensions.FrameworkPrincipal)
+		addMessageTag("hostname", dimensions.Hostname)
+
+		for _, datapoint := range message.Datapoints {
+			t, err := parseTimestamp(datapoint.Timestamp)
+			if err != nil {
+				log.Errorf("Could not parse timestamp '%s': %v", datapoint.Timestamp, err)
+				continue
+			}
+			v, err := toFloat64(datapoint.Value)
+			if err != nil {
+				log.Error(err)
+				continue
+			}
+
+			datapointTags := []string{}
+			addDatapointTag := func(key, value string) {
+				datapointTags = append(datapointTags, fmt.Sprintf("%s:%s", key, value))
+			}
+			for name, value := range datapoint.Tags {
+				addDatapointTag(name, value)
+			}
+
+			metric := DDMetric{
+				Metric: datapoint.Name,
+				Points: []DDDataPoint{{float64(t.Unix()), v}},
+				Tags:   append(messageTags, datapointTags...),
+				Unit:   datapoint.Unit,
+				Host:   host,
+			}
+			series.Series = append(series.Series, metric)
+		}
+	}
+
+	return series
+}
+
+func toFloat64(value interface{}) (float64, error) {
+	switch value := value.(type) {
+	case float64:
+		return value, nil
+	case float32:
+		return float64(value), nil
+	case int64:
+		return float64(value), nil
+	case int32:
+		return float64(value), nil
+	case int:
+		return float64(value), nil
+	case string:
+		return strconv.ParseFloat(value, 64)
+	}
+	return math.NaN(), fmt.Errorf("Could not convert %+v (%T) to float64", value, value)
+}
+
+func parseTimestamp(timestamp string) (*time.Time, error) {
```

Comment:
Likewise, since this logic is now duplicated between two pending plugins, should we also move this into the `plugin` package and export it?